docs/content/manual/manual.md

```diff
@@ -0,0 +1,5424 @@
+{}
+{"a":0,"b":0}
+~~~
+
+:::
+
+## Path expressions
+
+We now show which kinds of filters are path expressions, i.e.
+which filters can be used on the left-hand side of assignments.
+
+The following filters are path expressions:
+
+* `.` (identity)
+* `..` (recursive descent)
+* compound path: if it starts with some `f`, then `f` must be a path expression
+  * (`.[]` is a path expression because it starts with `.`, which is a path expression)
+  * (`{}[]` is _not_ a path expression, because it starts with `{}`, which is no path expression)
+* `if i then t else e end`: if `i` and `e` are path expressions
+* `f as $x | g`: if `g` is a path expression
+* `f, g`: if `f` and `g` are path expressions
+* `f | g`: if `f` and `g` are path expressions
+* `f // g`: if `f` and `g` are path expressions
+* `f?`: if `f` is a path expression
+* `label $x | f`: if `f` is a path expression
+* `break $x`
+* `def f: g; h` (function definition): if `h` is a path expression
+
+On the contrary, the following filters output values which do
+_not_ point to a part of their input, therefore they are
+_no_ path expressions:
+
+* [new values](#types-and-values), e.g. `1`, "Hello world", `[1, 2]`, `{a: 1}`
+* [arithmetic and comparison operations](#arithmetic-and-comparison), e.g. `. + 1`
+* `and`, `or`
+* `$x` (variable)
+* assignment (`|=`, `=`, `+=`, ...)
+
+For function calls, it depends on the function:
+If the function is implemented by definition and
+its definition is a path expression, then
+the function call is a path expression as well.
+For example, this is the case for [select](#select)) and [recurse](#recurse).
+However, most [builtin functions](#builtin-functions) return
+outputs that do not point to a part of their input,
+so calls to them are no path expressions.
+
+::: Note
+This characterisation of path expressions is an _underapproximation_;
+that is, there are filters that do not correspond to these criteria,
+yet they can be used on the left-hand side of assignments.
+For example, our criteria do not say that the filter
+`if true then empty else 0 end` is a path expression,
+because `0` is not a path expression.
+Despite this, we can happily use this filter on the left-hand side of an assignment.
+Such an assignment will always return its input, because
+`if true then empty else 0 end` always evaluates to `empty`,
+so jq does not attempt to evaluate `0` as path.
+:::
+
+::: Compatibility
+
+jaq's approach to handling assignments is quite different from that of jq and gojq.
+Specifically, jaq executes assignments without constructing compound paths.
+This means that jaq does not allow certain filters on the left-hand side of assignments,
+notably `f?` and `label $x | f`.
+jaq's approach is generally more performant, but in certain scenarios,
+jaq and jq will produce different results, in particular when using `f |= empty`.
+However, for the examples in this section, jq and jaq yield the same outputs.
+
+:::
+
+
+# Managing large programs
```

Comment:
Maybe in the future this could also include a section about debug techniques etc?

Reply:
That would be a nice idea.